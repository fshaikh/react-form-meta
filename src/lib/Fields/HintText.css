.hint-text {
    color: #737373;
    margin-top: 5px;
    margin-bottom: 10px;
    display: block;
}