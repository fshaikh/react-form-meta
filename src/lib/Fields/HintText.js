import React from 'react';
import PropTypes from 'prop-types';
import './HintText.css'

/**
 * Presentational component for showing hint text for a form field
 * @param {*} hintText - Hint text to show under a form field 
 */
const HintText = ({hintText}) => {
    return (
        <p className="hint-text">
            {hintText}
        </p>
    );
};

// Define Prop types
HintText.propTypes = {
    hintText: PropTypes.string
};

// exports go here
export default HintText;

