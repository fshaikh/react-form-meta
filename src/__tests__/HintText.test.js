import React from 'react';
import renderer from 'react-test-renderer';

import HintText from '../lib/Fields/HintText';

describe('Hint Text UI',() => {
    describe('Component tests for Hint Text UI', () => {
        it('should validate prop types',()=>{
            renderer.create(<HintText hintText={''}/>);
        });
    });
    describe('Snapshot tests for Hint Text UI', () => {
        it('should render Hint Text UI', () => {
            const field = {
                hintText: 'Password must be greater than 10 characters'
            };
            const tree = renderer.create(<HintText hintText={field.hintText} />).toJSON();
            expect(tree).toMatchSnapshot();
        });
    });
});