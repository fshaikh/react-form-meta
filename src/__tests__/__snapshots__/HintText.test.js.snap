// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hint Text UI Snapshot tests for Hint Text UI should render Hint Text UI 1`] = `
<p
  className="hint-text"
>
  Password must be greater than 10 characters
</p>
`;
